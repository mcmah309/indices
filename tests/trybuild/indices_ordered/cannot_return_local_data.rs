use indices::indices_ordered;

fn main() {
    one();
    two();
    three();
    four();
    five();
}

fn one() -> &'static mut i32 {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (four,) = indices_ordered!(slice, 4);
    four
}

fn two() -> (&'static mut i32, &'static mut i32) {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, four) = indices_ordered!(slice, 1, 4);
    (one, four)
}

fn three() -> &'static mut i32 {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, two, four) = indices_ordered!(slice, 1, 2, 4);
    one
}

fn four() -> (&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32) {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, two, three, four) = indices_ordered!(slice, 1, 2, 3, 4);
    (one, two, three, four)
}

fn five() -> (&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32) {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (zero, one, two, three, four) = indices_ordered!(slice, 0, 1, 2, 3, 4);
    (zero, one, two, three, four)
}
