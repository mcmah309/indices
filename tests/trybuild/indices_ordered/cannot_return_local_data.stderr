error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/indices_ordered/cannot_return_local_data.rs:15:5
   |
13 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
14 |     let (four,) = indices_ordered!(slice, 4);
15 |     four
   |     ^^^^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/indices_ordered/cannot_return_local_data.rs:22:5
   |
20 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
21 |     let (one, four) = indices_ordered!(slice, 1, 4);
22 |     (one, four)
   |     ^^^^^^^^^^^ returns a value referencing data owned by the current function

warning: unused variable: `two`
  --> tests/trybuild/indices_ordered/cannot_return_local_data.rs:28:15
   |
28 |     let (one, two, four) = indices_ordered!(slice, 1, 2, 4);
   |               ^^^ help: if this is intentional, prefix it with an underscore: `_two`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `four`
  --> tests/trybuild/indices_ordered/cannot_return_local_data.rs:28:20
   |
28 |     let (one, two, four) = indices_ordered!(slice, 1, 2, 4);
   |                    ^^^^ help: if this is intentional, prefix it with an underscore: `_four`

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/indices_ordered/cannot_return_local_data.rs:29:5
   |
27 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
28 |     let (one, two, four) = indices_ordered!(slice, 1, 2, 4);
29 |     one
   |     ^^^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/indices_ordered/cannot_return_local_data.rs:36:5
   |
34 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
35 |     let (one, two, three, four) = indices_ordered!(slice, 1, 2, 3, 4);
36 |     (one, two, three, four)
   |     ^^^^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/indices_ordered/cannot_return_local_data.rs:43:5
   |
41 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
42 |     let (zero, one, two, three, four) = indices_ordered!(slice, 0, 1, 2, 3, 4);
43 |     (zero, one, two, three, four)
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
